current.go

```diff
@@ -35,3 +38,51 @@ func (r *VaultRole) Exists(config ExistsConfig) (bool, error) {
 
 	return false, nil
 }
+
+type role struct {
+	AllowBareDomains bool          `json:"allow_bare_domains"`
+	AllowSubdomains  bool          `json:"allow_subdomains"`
+	AllowedDomains   string        `json:"allowed_domains"`
+	Organizations    string        `json:"organization"` // NOTE the singular form here.
+	TTL              time.Duration `json:"TTL"`
+}
```

Comment:
We have to deal with our own type because of how Vault represents data and because of how we work with it. One detail is that Vault holds allowed domains but our interface is about alternative names which contain the common name of a guest cluster.